File: 2446-determine-if-two-events-have-conflicts.java
class Solution {
    public boolean haveConflict(String[] event1, String[] event2) {
        int s1 = convertToMinutes(Integer.parseInt(event1[0].split(":")[0]),Integer.parseInt(event1[0].split(":")[1]));
        int e1 = convertToMinutes(Integer.parseInt(event1[1].split(":")[0]),Integer.parseInt(event1[1].split(":")[1]));
        int s2 = convertToMinutes(Integer.parseInt(event2[0].split(":")[0]),Integer.parseInt(event2[0].split(":")[1]));
        int e2 = convertToMinutes(Integer.parseInt(event2[1].split(":")[0]),Integer.parseInt(event2[1].split(":")[1]));
        return s1 <= e2 && s2 <= e1;
        // return event1[0].compareTo(event2[1]) <= 0 &&
        //        event2[0].compareTo(event1[1]) <= 0;
    }
    private int convertToMinutes(int h, int m){
        return (h * 60) + m;
    }
}

File: 2446-determine-if-two-events-have-conflicts.py
from typing import List
class Solution:
    def haveConflict(self, event1: List[str], event2: List[str]) -> bool:
        start1 = self.convertToMinutes(event1[0])
        end1 = self.convertToMinutes(event1[1])
        start2 = self.convertToMinutes(event2[0])
        end2 = self.convertToMinutes(event2[1])
        return start1 <= end2 and start2 <= end1

    def convertToMinutes(self, h : str) -> int:
        h, m = map(int,h.split(":"))
        return h*60 + m

        # start1, end1 = event1
        # start2, end2 = event2
        # return start1 <= end2 and start2 <= end1